src/qibo/callbacks.py

```diff
@@ -242,13 +242,19 @@ def __init__(self, hamiltonian: "hamiltonians.Hamiltonian"):  # type: ignore
 
     def apply(self, backend, state):
         assert type(self.hamiltonian.backend) == type(backend)
-        expectation = self.hamiltonian.expectation(state)
+        if state.__class__.__name__ == "Circuit":
```

Comment:
I'm assuming this is needed because importing the `Circuit` class would create a circular import?